src/stretchy.js

```diff
@@ -0,0 +1,274 @@
+};
+
+const svgSpan = function(group, options) {
+    // Create a span containing the SVG background-image
+
+    const label = group.value.label.substr(1);
+    let height = 0;
+    let depth = 0;
+    const classArray = ["stretchy"];
+    let fileName = "";
+
+    if (!document.implementation.hasFeature(
+        "http://www.w3.org/TR/SVG11/feature#BasicStructure", "1.1")) {
+
+        // SVG is not supported.
+        height = (label.substr(0, 1) === "x" ? 0.17 : 0.34);
+        depth  = (label.substr(0, 1) === "x" ? 0.17 : 0);
+
+    } else if (utils.contains(["widehat", "widetilde", "undertilde"], label)) {
+
+        // There are four SVG images available for each function.
+        // Choose a taller image when there are more characters.
+        const numChars = group.value.value.length;
+        if (numChars > 5) {
+            height = 0.34;
+            fileName = (label === "widehat" ? "widehat" : "tilde") + "4";
+        } else {
+            const imgIndex = [1, 1, 2, 2, 3, 3][numChars];
+            if (label === "widehat") {
+                height = [0, 0.16, 0.23, 0.23, 0.28, 0.28][numChars];
+                fileName = "widehat" + imgIndex;
+            } else {
+                height = [0, 0.15, 0.195, 0.195, 0.26, 0.26][numChars];
+                fileName = "tilde" + imgIndex;
+            }
+        }
+
+    } else {
+        const imgData = katexImagesData[label];
+        height = imgData[0];
+        depth = imgData[1];
+        fileName = imgData[2];
+
+        if (label.substr(0, 1) === "x") {
+            classArray.push("x-arrow");
+        } else if (utils.contains(["overbrace", "underbrace"], label)) {
+            classArray.push("horiz-brace");
+        } else if (utils.contains(["overbracket", "underbracket"], label)) {
+            classArray.push(label);
+        }
+    }
+
+    const node = buildCommon.makeSpan(classArray, [], options);
+    node.maxFontSize = 1.0;
+
+    if (utils.contains(["overbracket", "underbracket"], label)) {
+        if (group.value.thickness) {
+            const lineWt = group.value.thickness.value + "em";
+            node.style.borderWidth = (group.value.isOver ? lineWt : 0) + " " +
+                lineWt + " " + (group.value.isOver ? 0 : lineWt) + " " + lineWt;
+        }
+
+        if (group.value.height) {
+            height = group.value.height;
+            if (height > 1.0) {
+                node.maxFontSize = 1.1 * height;
+            }
+        }
+
+        if (height !== 0.3) {
+            node.style.height = height + "em";
+        }
+
+    } else if (!options.color) {
+        node.style.backgroundImage = "url(" + katexImgPath() +
+            fileName + ".svg)";
+        // By writing that image path inline, we eliminate ~40 CSS classes.
+        if (height + depth !== 0.34) {
+            node.style.height = (height + depth) + "em";
+        }
+    } else {
+        // Add color. Use CSS mask, if possible
+        if (maskIsSupported()) {
```

Comment:
If there isn't a good one-size-fits-all solution, it seems like the best thing to do would be to add an option that lets you decide between "work on Edge" vs. "work fast", when you can't autodetect it.

I'm a little confused why you need an XMLHttpRequest, though.  Is it not worth just inlining the SVG into the JavaScript code (const mySVG = "...") and using that as needed?  I remember we discussed this before, but why use external files at all, when you can embed it all in the JavaScript source?